centreon/src/Core/ServiceTemplate/Domain/Model/NewServiceTemplate.php

```diff
@@ -147,11 +147,10 @@ public function __construct(
      */
     public function setName(string $name): void
     {
-        $name = preg_replace('/\s{2,}/', ' ', $name);
+        $name = ServiceTemplate::formatName($name);
```

Comment:
`$name` is th parameter name of the method dans is a `string`
But `::formatName()` can return `null|string`, then we can have a `null` inside the original `$name` variable.
It can be weird, I suggest something like that to avoid misunderstanding : 
```suggestion
        $name = ServiceTemplate::formatName($name)
            ?? throw AssertionException::notNull($this->className . '::name');
```

Reply:
I don't see where there's any misunderstanding, as the condition `$name === null` is, in my opinion, sufficiently explicit.